tests/unit-ext/test-ext-args.c

```diff
@@ -0,0 +1,298 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Unit test for jerry-ext/args.
+ */
+
+#include "jerryscript.h"
+#include "jerryscript-ext/args.h"
+#include "test-common.h"
+
+#include <string.h>
+
+const char *test_source = (
+                           "var arg1 = true;"
+                           "var arg2 = 10.5;"
+                           "var arg3 = 'abc';"
+                           "var arg4 = function foo() {};"
+                           "test_validator1(arg1, arg2, arg3, arg4);"
+                           "arg1 = new Boolean(true);"
+                           "arg3 = new String('abc');"
+                           "test_validator1(arg1, arg2, arg3);"
+                           "arg2 = new Number(10.5);"
+                           "test_validator1(arg1, arg2, arg3);"
+                           "test_validator1(arg1, 10.5, 'abcdef');"
+                           "var obj_a = new MyObjectA();"
+                           "var obj_b = new MyObjectB();"
+                           "test_validator2.call(obj_a, 5);"
+                           "test_validator2.call(obj_b, 5);"
+                           "test_validator2.call(obj_a, 1);"
+                           );
+
+
+
+static const jerry_object_native_info_t thing_a_info =
+{
+  .free_cb = NULL
+};
+
+static const jerry_object_native_info_t thing_b_info =
+{
+  .free_cb = NULL
+};
+
+typedef struct
+{
+  int x;
+} my_type_a_t;
+
+typedef struct
+{
+  bool x;
+} my_type_b_t;
+
+static my_type_a_t my_thing_a;
+static my_type_b_t my_thing_b;
+
+
+static int validator1_count = 0;
+static int validator2_count = 0;
+
+/**
+ * The handler should have following arguments:
+ *   this: Ignore.
+ *   arg1: Bool.
+ *   arg2: Number. It must be strict primitive number.
+ *   arg3: String.
+ *   arg4: function. It is an optional argument.
+ *
+ */
+static jerry_value_t
+test_validator1_handler (const jerry_value_t func_obj_val __attribute__((unused)), /**< function object */
+                         const jerry_value_t this_val, /**< this value */
+                         const jerry_value_t args_p[], /**< arguments list */
+                         const jerry_length_t args_cnt) /**< arguments length */
+{
+
+  bool arg1;
+  double arg2 = 0.0;
+  char arg3[5];
+  jerry_value_t arg4 = jerry_create_undefined ();
+
+  jerryx_arg_t mapping[] =
+  {
+    /* ignore this */
+    jerryx_arg_ignore (),
+    /* 1st argument should be boolean */
+    jerryx_arg_boolean (&arg1, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    /* 2nd argument should be strict number */
+    jerryx_arg_number (&arg2, JERRYX_ARG_NO_COERCE, JERRYX_ARG_REQUIRED),
+    /* 3th argument should be string */
+    jerryx_arg_string (arg3, 5, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    /* 4th argument should be function, and it is optional */
+    jerryx_arg_function (&arg4, JERRYX_ARG_OPTIONAL)
```

Comment:
> allocates a big structure on the stack 

I thought it might be possible to make the `jerryx_arg_t mapping[]` a `static const` to avoid the stack issue, because the elements are actually all constant data.
Unfortunately the compiler isn't smart enough: `error: initializer element is not a compile-time constant`.
I guess if we went with macros instead of inline functions to create the `jerryx_arg_t`s, making it `static const` would be possible.